src/dotnet/Streaming.UI.Blazor/Components/AudioRecorder/workers/audio-vad-worker.ts

```diff
@@ -147,19 +131,15 @@ async function processQueue(): Promise<void> {
 
             const buffer = queue.shift();
             let vadEvent: VoiceActivityChange | number;
-            const hasNNVad = nnVoiceDetector != null && resampler != null;
+            const hasNNVad = nnVoiceDetector != null;
 
             // might be switched to NN Vad, but still has queue with wrong buffers
             if (hasNNVad && buffer.byteLength == SAMPLES_PER_WINDOW_32 * 4) {
-                const dataToResample = new Uint8Array(buffer);
-                const resampled = resampler.processChunk(dataToResample, resampleBuffer);
-                // 32 ms at 16000 Hz
-                const monoPcm = new Float32Array(resampled.buffer, 0, 512);
+                const monoPcm = new Float32Array(buffer, 0, SAMPLES_PER_WINDOW_32);
                 vadEvent = await nnVoiceDetector!.appendChunk(monoPcm);
             }
             else if (buffer.byteLength == SAMPLES_PER_WINDOW_30 * 4) {
-                // 30 ms at 48000 Hz
-                const monoPcm = new Float32Array(buffer, 0, 1440);
+                const monoPcm = new Float32Array(buffer, 0, SAMPLES_PER_WINDOW_30);
                 vadEvent = await webrtcVoiceDetector.appendChunk(monoPcm);
             }
```

Comment:
What will happen if none of above conditions are met?

Reply:
we will have to skip VAD processing for wrong sample buffers